### PR TITLE
Removes account from AWS image and named image cache key.

### PR DESCRIPTION
**Pull request by cfieber**
AMI Id uniquely identifies an image within a region. This change just hardcodes a value where previously
we would put the account name (allows for compatible key parsing between versions older than this change).

Images are now only cached by the image caching agent if they are owned by that agent's account.

@spinnaker/netflix-reviewers PTAL

### CODE REVIEW

**Comment by ajordens** [comment]
Needs a rebase if we're still thinking this would be beneficial.

**Comment by anotherchrisberry** [comment]
ping

**Comment by cfieber** [comment]
notdeadyet

**Comment by ajordens** [comment]
Thoughts on reviving this thing ... looks close.

Would complete my image tagging work (as I'd really only like to tag in the owning account)

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/ImageCachingAgent.groovy [hunk 1]
```diff
@@ -102,10 +102,10 @@ class ImageCachingAgent implements CachingAgent, AccountAware, DriftMetric {
     Collection<CacheData> namedImageCacheData = new ArrayList<>(images.size())
 
     for (Image image : images) {
-      if (includePublicImages || !image.public) {
+      if (includePublicImages || !image.public || image.ownerId == account.accountId) {
```
Comment (ajordens) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/ImageCachingAgent.groovy]: Might need some love in the case where `clouddriver` isn't indexing the owning account.

**Comment by cfieber** [comment]
Yeah I want to get this in - need to do some testing mainly to make sure things behave ok (and it was waiting for the dual mode redis to handle to key format change)

**Comment by ajordens** [comment]
Bump bump.

**Comment by ewiseblatt** [comment]
please reap this.

**Pull request closed without merging**